test/auth_cluster/test_password_login_message.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (C) 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+from test.pylib.manager_client import ManagerClient
+import pytest
+import logging
+from cassandra.cluster import NoHostAvailable
+
+logger = logging.getLogger(__name__)
+
+"""
+Tests the error message after half of the cluster goes down.
+
+This test reproduces ScyllaDB OSS issue #2339. We create a
+new cluster with 2 nodes and then stop one of them. We then
+try to connect to the cluster and expect to get the error
+message:
+"Cannot achieve consistency level for cl QUORUM. Requires 1, alive 0"
+"""
+@pytest.mark.asyncio
+async def test_login_message_after_half_of_the_cluster_is_down(manager: ManagerClient) -> None:
+    """Tests the error message after half of the cluster goes down"""
+    config = {
+        'failure_detector_timeout_in_ms': 2000,
+        'authenticator': 'PasswordAuthenticator',
+        'authorizer': 'CassandraAuthorizer',
+        'permissions_validity_in_ms': 0,
+        'permissions_update_interval_in_ms': 0,
+    }
+    
+    servers = [await manager.server_add(config=config) for _ in range(2)]
+    cql = manager.get_cql()
+    cql.execute(f"ALTER KEYSPACE system_auth WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': 3}}")
+    await manager.server_stop_gracefully(servers[1].server_id)
+    logger.info(f"Waiting until server {servers[0]} marks {servers[1]} as dead")
+    await manager.server_not_sees_other_server(servers[0].ip_addr, servers[1].ip_addr)
```

Comment:
I'd double-check if `server_not_sees_other_server` is really needed if we already execute `server_stop_gracefully` above. I assume that if 2nd node is stopped gracefully, the 1st node shouldn't see it.

Reply:
Stopping a node gracefully is one thing, the others noticing it as down is another. Before that, they might still attempt to contact it and you will get a timeout, instead of a no host available error.

Reply:
OK, so better to leave it, thanks.

Reply:
It seems to work reliably without it, so I'll remove it.